### PR TITLE
Add ssd

### PR DESCRIPTION
**Pull request by heavengate**
**Add SSD & Refine DataLoader**
- add `VOC` dataset and evaluation
- add SSD model: ssd_vgg16_300_240e_voc.yml
- refine `DataLoader` yield data in dictionary format to simplify `train/eval/infer`, `export_model` test pass in dictionary format
- SSD export model and deploy infer ok

### SSD on Pascal VOC

| 骨架网络        | 网络类型       | 每张GPU图片个数 | 学习率策略 |推理时间(fps) | Box AP |                           下载                          | 配置文件 |
| :-------------- | :------------- | :-----: | :-----: | :------------: | :-----: | :-----------------------------------------------------: | :-----: |
| VGG             | SSD            |    8    |   240e    |     ----     |  78.2  | [下载链接](https://paddlemodels.bj.bcebos.com/object_detection/dygraph/ssd_vgg16_300_240e_voc.pdparams) | [配置文件](https://github.com/PaddlePaddle/PaddleDetection/tree/dygraph/configs/ssd_vgg16_300_240e_voc.yml) |

training on 4 * P40 for 19h

### CODE REVIEW

**Review by wangxinxin08 — APPROVED** [review]
LGTM

**Review by qingqing01 — APPROVED** [review]

File: ppdet/modeling/ops.py [hunk 1]
```diff
@@ -1254,6 +1254,178 @@ def matrix_nms(bboxes,
...
+    idx_rank = loss_idx.argsort(axis=1)
+    num_negs = []
+    for i in range(matched_indices.shape[0]):
+        cur_idx = loss_idx[i]
+        cur_num_pos = num_pos[i]
+        num_neg = paddle.clip(cur_num_pos * neg_pos_ratio, max=pos.shape[1])
+        num_negs.append(num_neg)
+    num_neg = paddle.stack(num_negs, axis=0).expand_as(idx_rank)
+    neg_mask = (idx_rank < num_neg).astype(conf_loss.dtype)
+    return neg_mask
```
Comment (qingqing01) [inline ppdet/modeling/ops.py:1426]: label_target_assign, bbox_target_assign, mine_hard_example这几个移动到ssd-loss那块就行吧

**Review by qingqing01 — COMMENTED** [review]
没看到ssd_loss文件

File: ppdet/modeling/ops.py (hunk 1 above)
Comment (heavengate) [reply ppdet/modeling/ops.py]: Done, thanks!  `ssd_loss.py` pushed

**Review by qingqing01 — APPROVED** [review]

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**